pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagent.go

```diff
@@ -211,6 +217,23 @@ func installer(allWorkers bool, secretName string) []runtime.Object {
 			},
 		},
 	}
+
+	if generation != "0" {
```

Comment:
After talking to @dweomer , we can use secretRef to force job rerun. This doesn't need SUC pr to support that anymore.